src/guide/extras/composition-api-faq.md

```diff
@@ -2,156 +2,164 @@
 
-<img alt="folder component before" src="./images/options-api.png" width="129" height="500" style="margin: 1.2em auto">
+<img alt="versione precedente del componente" src="./images/options-api.png" width="129" height="500" style="margin: 1.2em auto">
 
-Notice how code dealing with the same logical concern is forced to be split under different options, located in different parts of the file. In a component that is several hundred lines long, understanding and navigating a single logical concern requires constantly scrolling up and down the file, making it much more difficult than it should be. In addition, if we ever intend to extract a logical concern into a reusable utility, it takes quite a bit of work to find and extract the right pieces of code from different parts of the file.
+Si noti come il codice che si occupa della stessa logica funzionale sia costretto a essere suddiviso in diverse options, localizzate in diverse parti del file. In un componente che ha centinaia di linee di codice, capire e navigare ogni singola logica funzionale richiede costantemente di scrollare su e giù, rendendolo la comprensione molto più difficoltosa di quanto dovrebbe essere. Inoltre, se vogliamo estrarre una logica funzionale esponendola come utility riutilizzabile, questo richiede un pò di lavoro per trovare ed estrarre i pezzi di codice posizionate in parti differenti del file.
 
-Here's the same component, before and after the [refactor into Composition API](https://gist.github.com/yyx990803/8854f8f6a97631576c14b63c8acd8f2e):
 
-![folder component after](./images/composition-api-after.png)
+Qui lo stesso componente, prima e dopo il [refactor tramite Composition API](https://gist.github.com/yyx990803/8854f8f6a97631576c14b63c8acd8f2e):
 
-Notice how the code related to the same logical concern can now be grouped together: we no longer need to jump between different options blocks while working on a specific logical concern. Moreover, we can now move a group of code into an external file with minimal effort, since we no longer need to shuffle the code around in order to extract them. This reduced friction for refactoring is key to the long-term maintainability in large codebases.
+![versione refactor del componente](./images/composition-api-after.png)
 
-### Better Type Inference {#better-type-inference}
+Si noti come il codice relativo alla stessa logica funzionale può ora essere raggruppato: non abbiamo più bisogno di saltare tra blocchi di options mentre lavoriamo su una logica funzionale specifica. Infine, possiamo muovere un gruppo di codice in un file esterno con il minimo effort, poiché non abbiamo più bisogno di muoverci nel codice per estrarre la parte che ci interessa.  La riduzione degli impatti in caso di refactoring è la chiave per la manutenibilità a lungo termine in progetti molto complessi.
```

Comment:
possiamo muovere un gruppo ->
possiamo spostare un gruppo

La riduzione degli impatti in caso di refactoring è la chiave per la manutenibilità a lungo termine in progetti molto complessi. ->
La riduzione dell'attrito al refactoring è la chiave per la manutenibilità a lungo termine in progetti complessi.

Reply:
Sinceramente la cambierei:

La riduzione degli impatti in caso di refactoring è la chiave per la manutenibilità a lungo termine in progetti molto complessi. ->
La riduzione dell'attrito al refactoring è la chiave per la manutenibilità a lungo termine in progetti complessi.
io propongo:
Questa difficoltà ridotta al refactoring è la chiave per la manutenibilità a lungo termine in progetti complessi.

Reply:
yep, volendo non essere troppo letterali potremmo renderla più naturale in Italiano con : 
**Favorire/agevolare il refactoring ...**